### PR TITLE
add segment to registrar

### PR DESCRIPTION
**Pull request by jansenk**
basic setup for segment.io integration with registrar

[EDUCATOR-4131](https://openedx.atlassian.net/browse/EDUCATOR-4131)

related: https://github.com/edx/configuration/pull/5057

### CODE REVIEW

File: registrar/apps/api/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"""
+Initialization file for API app. Sets up config for segment.io integration
+"""
+default_app_config = 'registrar.apps.api.apps.ApiConfig'
```
Comment (schenedx) [inline registrar/apps/api/__init__.py]: According to latest django documentation at https://docs.djangoproject.com/en/2.1/ref/applications/#configuring-applications (see paragraph 5), you should not have the changes in here to incorporate a new app config.

File: registrar/settings/base.py [hunk 2]
```diff
@@ -233,3 +234,6 @@
 
 # Set up logging for development use (logging to stdout)
 LOGGING = get_logger_config(debug=DEBUG, dev_env=True, local_loglevel='DEBUG')
+
+# API key for segment.io event tracking
+SEGMENT_KEY = "replace-me"
```
Comment (schenedx) [inline registrar/settings/base.py]: new line please

File: registrar/apps/api/apps.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+class ApiConfig(AppConfig):
+    name = 'registrar.apps.api'
+    verbose_name = 'API'
+
+    def ready(self):
+        """
+        Initialize Segment analytics module by setting the write_key.
+        """
+        if settings.SEGMENT_KEY:
+            analytics.write_key = settings.SEGMENT_KEY
```
Comment (schenedx) [inline registrar/apps/api/apps.py:18]: Are you sure the `settings.SEGMENT_KEY` won't be None? Should we handle the `None` case

**Review by schenedx — CHANGES_REQUESTED** [review]
Mostly looks good. Let me know if you have feedback

File: registrar/apps/api/__init__.py (hunk 1 above)
Comment (jansenk) [reply registrar/apps/api/__init__.py]: hmm, okay. I was following the segment django documentation.
https://segment.com/docs/sources/server/python/#django

**Review by schenedx — APPROVED** [review]
👍

**Review by iloveagent57 — APPROVED** [review]
That looks right to me 👍

**Pull request merged**